src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -267,9 +281,19 @@ private static string BuildConfigFileName()
             return GetAppName() + ConfigFileSuffix;
         }
 
-        private static string BuildTraceFileName()
+        private static string BuildTraceFileName(bool multiFile)
```

Comment:
I really don't like this wacky invariant that the TraceFile name includes the suffix when you are doing single file but DOES included it for multi-file.    I want it to not be conditional.   You can do it either way but my bias is to make the 'common' case simple and use names that mean what they say.  FileNames are normally the whole path, so my recommendation is that you do this uniformly.    Then when you need multiple files, you simply put it 'before' the netperf suffix (if Present), or simply add it as a suffix (if the .netperf is not there).     There are other variations but the key is that if you don't care about mulit-file (which is the common case, everything is what you expect. You have pushed all the complexity into the Multi-file case.

Reply:
Yeah, I like the idea of just pushing the "file numbering" logic down entirely to EventPipe. If `s_nextFileIndex` is `0`, the file name is left unmodified, otherise the number is injected before the `netperf` suffix (the managed initialization code can ensure the input string either always has or never has the `.netperf` suffix, whichever is easier for working with in the native code). Then the file naming logic is consistent and just tied to the number of times the file is rolled. A "single-file" tracing session would just be one where the roll interval is "infinite" and so `s_nextFileIndex` is always `0`.

That's all internal details to us though so I'm not worried either way :)